opentelemetry-collector-releases: init at 0.114.0

zimbatm opened the pull request:
## Motivation

 `opentelemetry-collector-contrib` is stuck on an old version because they removed the main.go in version 0.111.0 and later. Before that, the binary folder was saying:

> This folder contains the sources for the `otelcontribcol` test binary. This binary is intended for internal **TEST PURPOSES ONLY**. The source files in this folder are **NOT** the ones used to build any official OpenTelemetry Collector releases.

`opentelemetry-collector` is still embedding the main.go, but it has the folder has the same warning as above.

So, I took some time rebuilding all of this to follow more closely what upstream is doing.

## Things done

Re-packaged everything using their "ocb" build tool.

<!-- Please check what applies. Note that these are not hard requirements but merely serve as information for reviewers. -->

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [25.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) (or backporting [24.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) and [25.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2505.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

<!--
To help with the large amounts of pull requests, we would appreciate your
reviews of other pull requests, especially simple package updates. Just leave a
comment describing what you have tested in the relevant package/service.
Reviewing helps to reduce the average time-to-merge for everyone.
Thanks a lot if you do!

List of open PRs: https://github.com/NixOS/nixpkgs/pulls
Reviewing guidelines: https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#reviewing-contributions
-->

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

zimbatm committed:
opentelemetry-collector-builder: 0.112.0 -> 0.114.0

zimbatm committed:
opentelemetry-collector-releases: init at 0.114.0

zimbatm committed:
opentelemetry-collector: alias to opentelemetry-collector-releases.otelcol

zimbatm committed:
opentelemetry-collector-contrib: alias to opentelemetry-collector-releases.otelcol-contrib

zimbatm commented:
/cc @uri-canva and @06kellyjac as current maintainers.

/cc @flokli as per today's conversation.

flokli commented on pkgs/by-name/op/opentelemetry-collector-builder/package.nix:
```diff
@@ -5,17 +5,19 @@
-  version = "0.112.0";
+  # Also update `pkgs/tools/misc/opentelemetry-collector/releases.nix`
+  # whenever that version changes.
+  version = "0.114.0";
```
Is there a way to keep the hashes closer to this version number?

Theres *two* places with hashes in `releases.nix`, pretty far apart and it's probably only a matter of time until some are forgotten to be updated.

zimbatm replied:
It might be worth moving this package out of pkgs/by-name and have it close to the rest.

The next bigger work is to write a shared updater script.

flokli commented on pkgs/tools/misc/opentelemetry-collector/releases.nix:
```diff
@@ -0,0 +1,172 @@
+
+      outputHash = hash;
+      outputHashMode = "recursive";
+      outputHashAlgo = if hash == "" then "sha256" else null;
```
```suggestion
      outputHashAlgo = null;
```


we always call this with a hash that is not null.

This conditional logic only made sense in places where we didn't pass an SRI hash and had to explicitly communicate the algo separately.

zimbatm replied:
This allows empty hashes to be used to recalculate the FOD. It's less work than replacing the FOD with lib.fakeHash;

flokli replied:
Ah, ok then. Sorry für the noise.

flokli commented on pkgs/tools/misc/opentelemetry-collector/releases.nix:
```diff
@@ -0,0 +1,172 @@
+
+        # upstream strongly recommends disabling CGO
+        # additionally dependencies have had issues when GCO was enabled that weren't caught upstream
+        # https://github.com/open-telemetry/opentelemetry-collector/blob/main/CONTRIBUTING.md#using-cgo
```
This has gotten a stronger warning and was moved to https://github.com/open-telemetry/opentelemetry-collector/blob/9d2685f404ef65b46152ae80c36ff84f303e3cc6/docs/coding-guidelines.md?plain=1#L311 (permalink)

flokli replied:
side note: systemd journal ingestion seems to still work as the receiver shells out instead of liking against libsystemd, which would require CGO: https://github.com/open-telemetry/opentelemetry-collector-contrib/tree/main/receiver/journaldreceiver

flokli replied:
Can you update the link to the permalink from the first comment?

zimbatm commented:
Also /cc @DavSanchez as the maintainer of ocb for opinion. Are you using the open-collective-builder for anything yet?

DavSanchez commented:
> Also /cc @DavSanchez as the maintainer of ocb for opinion. Are you using the open-collective-builder for anything yet?

Hello @zimbatm ! Currently I only use `ocb` on private flakes, please feel free to move this PR forward 👌

zimbatm committed:
opentelemetry-collector-builder: move to the package set

Since the package versions are best kept in sync, move it close to the
other opentelemetry-collector packages.

zimbatm commented:
Ok, I moved the builder next to the rest and will stop there for now. It's already better than what we had before.

A follow-up work would be to build the updater script. I'll be motivated to work on that once the updates become annoying.

zimbatm commented:
## `nixpkgs-review` result

Generated using [`nixpkgs-review`](https://github.com/Mic92/nixpkgs-review).

Command: `nixpkgs-review pr 357386`

---
### `x86_64-linux`
<details>
  <summary>:white_check_mark: 5 packages built:</summary>
  <ul>
    <li>opentelemetry-collector</li>
    <li>opentelemetry-collector-builder</li>
    <li>opentelemetry-collector-contrib</li>
    <li>opentelemetry-collector-releases.otelcol-k8s</li>
    <li>opentelemetry-collector-releases.otelcol-otlp</li>
  </ul>
</details>

stratdev3 commented:
the current [v0.112 in nixos 24.11 is broken](https://github.com/NixOS/nixpkgs/issues/368321).

Could it be possible to update the latest version to nixos 24.11 ?

Thanks
